benthomasson/eda/tests/unit/event_source/test_webhook.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+import aiohttp
+import asyncio
+
+from plugins.event_source.webhook import main as webhook_main
+
+
+async def start_server(queue, args):
+    await webhook_main(queue, args)
+
+
+async def post_code(server_task, info):
+    url = f'http://{info["host"]}/{info["endpoint"]}'
+    payload = info["payload"]
+
+    async with aiohttp.ClientSession() as session:
+        async with session.post(url, json=payload) as resp:
+            print(resp.status)
```

Comment:
@hsong-rh Should we raise an exception here if the status is not 200.

Reply:
@mkanoor If we raise an exception here, when it really happens, the next line `server_task.cancel()` will not be called. The server will keep running and never back.